File: python/Algo/binary_search.py
def binary_search(list, searchItem):    
    if(len(list)==0):
        return False
    else:
        first = 0
        last = len(list) - 1
        while first <= last:
            midpoint = (first + last)//2
            if(list[midpoint] == searchItem):
                return True
            elif( list[midpoint] > searchItem):
                last = midpoint - 1 
            else:
                first = midpoint + 1
        return None

inputData = [1,3,5,6,7,8,9,10,12,14,16]

result = binary_search([], 11)

print("Search item", result)

File: RoBoGo/2023/HackerRank/binary_search.js


function binary_search(inputList, searchItem) {
    let first = 0;
    let last = inputList.length - 1;
    console.log('last ', last)
    while( parseInt(first) <= parseInt(last) ){
        const mid = Math.floor((parseInt(first) + parseInt(last))/2);
        console.log('first', first);
        console.log('mid', mid);
        console.log('last', last);
        console.log('searchItem', searchItem);
        console.log('inputList[mid]', inputList[mid]);

        if(inputList[mid] == searchItem){
            return true;
        }else if(inputList[mid] > searchItem) {
            last = mid-1;
        } else {
            first = mid + 1;
        }
    }
    return false;
}

let inputdata = [2,3,4,5,7,8,9,10,13,15];

console.log(binary_search(inputdata,50));